ui/scripts/vpc.js

```diff
@@ -745,9 +745,22 @@
                         }
                     },
                     dataProvider: function(args) {
+                        var array1 = [];
+                        if (args.filterBy != null) {
+                            if (args.filterBy.search != null && args.filterBy.search.by != null && args.filterBy.search.value != null) {
+                                switch (args.filterBy.search.by) {
+                                    case "name":
+                                        if (args.filterBy.search.value.length > 0)
+                                            array1.push("&keyword=" + args.filterBy.search.value);
+                                        break;
+                                }
+                            }
+                        }
                         $.ajax({
-                            url: createURL('listLoadBalancers'),
+                            url: createURL('listLoadBalancers' + array1.join("")),
                             data: {
+                                page: args.page,
+                                pageSize: pageSize,
                                 networkid: args.context.networks[0].id,
```

Comment:
Thanks @vedulasantosh, however, we can still improve the code, instead of putting the keys/values in an array, why not create a `data` object and push keys/values to it and directly give it to `$.ajax`. Furthermore, the nested `ifs and switch/case` can be clubbed together. Sorry to nitpick on this, your code is valid but it would be great if you can improve it further.